sdk/src/main/java/com/stormpath/sdk/ApiManager.java

```diff
@@ -0,0 +1,360 @@
+                    }
+
+                    successCallback(null);
+                } catch (Throwable t) {
+                    failureCallback(t);
+                }
+            }
+        });
+    }
+
+    public void refreshAccessToken(final StormpathCallback<Void> callback) {
+        String refreshToken = preferenceStore.getRefreshToken();
+
+        if (StringUtils.isBlank(refreshToken)) {
+            callbackExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    callback.onFailure(
+                            new IllegalStateException("refresh_token was not found, did you forget to login? See debug logs for details."));
+                }
+            });
+            return;
+        }
+
+        RequestBody formBody = new FormBody.Builder()
+                .add("refresh_token", refreshToken)
+                .add("grant_type", "refresh_token")
+                .build();
+
+        Request request = new Request.Builder()
+                .url(config.oauthUrl())
+                .header("Accept", "application/json")
+                .post(formBody)
+                .build();
+
+        okHttpClient.newCall(request).enqueue(new OkHttpCallback<Void>(callback) {
+            @Override
+            protected void onSuccess(Response response, StormpathCallback<Void> callback) {
+                try {
+                    LoginResponse loginResponse = moshi.adapter(LoginResponse.class).fromJson(response.body().source());
+                    if (StringUtils.isBlank(loginResponse.getAccessToken())) {
+                        failureCallback(new RuntimeException("access_token was not found in response. See debug logs for details."));
+                        return;
+                    }
+
+                    if (StringUtils.isBlank(loginResponse.getRefreshToken())) {
+                        Stormpath.logger().e("There was no refresh_token in the login response!");
+                    }
+
+                    preferenceStore.setAccessToken(loginResponse.getAccessToken());
+                    preferenceStore.setRefreshToken(loginResponse.getRefreshToken());
+                    successCallback(null);
+                } catch (Throwable t) {
+                    failureCallback(t);
+                }
+            }
+        });
+    }
+
+    public void getUserProfile(final StormpathCallback<UserProfile> callback) {
+        String accessToken = preferenceStore.getAccessToken();
+
+        if (StringUtils.isBlank(accessToken)) {
+            callbackExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    callback.onFailure(
+                            new IllegalStateException("access_token was not found, did you forget to login? See debug logs for details."));
+                }
+            });
+            return;
+        }
+
+        Request request = new Request.Builder()
```

Comment:
While this is very readable, the two header lines are repeated a lot. I wonder if as a convenience, you could create another `Builder` method that takes the accessToken and sets those two headers internally? Like:

```
Request request = new Request.Builder(accessToken)
    .url(config.userProfileUrl())
    .get()
    .build();
```

I'm not married to it - and I wouldn't hold up the MVP for it - just a thought.